src/internet_message.c

```diff
@@ -163,22 +160,26 @@ static int im_header_add_fold_ctx(struct im_header_context *ctx)
  */
 int im_header_feed(struct im_header_context *ctx, char c)
 {
+	jsval header;
+
 	switch (ctx->state) {
 	case IM_H_NAME1:
 		if (strchr(tab_space, c)) {
-			if (list_empty(ctx->hdrs))
-				return IM_PARSE_ERROR;
-			if (im_header_add_fold_ctx(ctx))
+			if (im_header_add_fold_ctx(ctx, &header))
 				return IM_OUT_OF_MEM;
 			if (ctx->curr_size++ >= ctx->max_size)
 				return IM_OVERRUN;
-			if (string_buffer_append_char(&ctx->sb, ' '))
+			if (string_buffer_append_char(&ctx->sb, c))
 				return IM_OUT_OF_MEM;
 			ctx->state = IM_H_FOLD;
 			return IM_OK;
 		}
-		if (!list_empty(ctx->hdrs) && im_header_set_value_ctx(ctx))
+		if (ctx->start && im_header_set_value_ctx(ctx, &header)) {
 			return IM_OUT_OF_MEM;
+		}
+
+		ctx->start = 1;
```

Comment:
Maybe it's better to initialize 'header' with JSVAL_NULL and then just check if it's not null. See the note above about the initialization of 'header'.